src/main/java/org/cactoos/text/TextOf.java

```diff
@@ -335,17 +315,7 @@ public TextOf(final String input) {
     /**
      * Ctor.
      *
-     * @param input The String
-     * @param cset The Charset
-     */
-    public TextOf(final String input, final Charset cset) {
-        this(() -> new String(input.getBytes(cset), cset));
-    }
-
-    /**
-     * Ctor.
      * @param iterable The iterable to convert to string
-     * @since 0.21
      */
     public TextOf(final Iterable<?> iterable) {
         this(
```

Comment:
@baudoliver7 for this one, as I was proposing in my previous comment, let's directly delegate to Joined without using a lambda, and let's have `TextOf(final Iterator<?> iterator)` delegate to the `Iterable` constructor directly

Reply:
@victornoel When we delegate `TextOf(final Iterator<?> iterator)` to the `Iterable` constructor directly, it changes code behaviour and fails this test :
```java
@Test
void readsIteratorToText() throws Exception {
    new Assertion<>(
        "Can't read Iterator to Text",
        new TextOf(
            new IteratorOfChars("qwer")
        ).asString(),
        new IsEqual<>("qwer")
    ).affirm();
}
```
with this error :
```cmd
[ERROR]   TextOfTest.readsIteratorToText:329 Can't read Iterator to Text
Expected: "qwer"
 but was: was "q, w, e, r"
```
FYI, that is what I changed in the first constructor :
```java
/**
 * Ctor.
 *
 * @param iterable The iterable to convert to string
 */
public TextOf(final Iterable<?> iterable) {
    super(
        new Joined(
            ", ",
            new Mapped<>(
                Object::toString,
                iterable
            )
        )
    );
}
```

Reply:
@baudoliver7 I think we should uniformise the behaviour of `TextOf` for `Iterator` and `Iterable`... that is something I overlooked when making previous comments :)
The best would be to add a todo and leave things as they are. But the todo should propose and idea of solution or at least be clear that it should be consistent with the fact that we already have `Concatenated` and `Joined`, so maybe we should only support `Iterable` and `Iterator` of `Character`?

Reply:
@victornoel I agree to support only `Iterable` and `Iterator` of `Character` ?

Reply:
@baudoliver7 indeed, that seems to be the best!

Reply:
@victornoel It's already supported in my last commit :)

Reply:
@baudoliver7 I meant to have support for `Iterable<Character>` and `Iterator<Character>`, both behaving similarly, most certainly the iterator one delegating to the iterable one

Reply:
@victornoel Use constructor for supporting `Iterable<Character>` fails most part of code. In fact, Constructor `Iterator<Character>` is use to form a word with a list of characters. However, Constructor `Iterable<?>` concat a list of string. Both behaviour are different.

Reply:
@baudoliver7 yes, the point of the proposed change is to modify actual behaviour of `TextOf` to simplify it. So it means that we need to modify other parts of cactoos that relied on this in order to preserve their behaviour (using joined our concatenated). Let's add a new todo for now to do it, ok?

Reply:
@victornoel Ok. I added a todo.